### ISSUE TITLE
Allocation failure

### ISSUE THREAD

**Issue by delebash** [duartealexf/sql-ddl-to-json-schema#43]
Windows 10 x64
node 11.13.0
sql-ddl-to-json-schema version 3.1.0

Attached is the sql file generated from a table.  When I try to run node .  I get an error regarding Allocation failure and memory. I have 32g of memory.

index.js
```

const Parser = require('sql-ddl-to-json-schema');

const { join } = require('path');
const { readFileSync } = require('fs');
const { writeFileSync } = require('fs');
const parser = new Parser('mysql');
const options = {};

let sql = readFileSync(join(__dirname, 'example', 'animal.sql')).toString();
sql = sql.replace( new RegExp("\\/\\*!.*[\\s\\S]|^.*--.*","gm"),"");

parser.feed(sql)
  .toJsonSchemaFiles(join(__dirname, 'example'), { extension: '.schema.json', indent: 4, useRef: false })
  .then(outputFilePaths => {
    console.log(JSON.stringify(outputFilePaths, null, 2));
  });

```

Below is the attached .sql file

[animal.sql.txt](https://github.com/duartealexf/sql-ddl-to-json-schema/files/3051346/animal.sql.txt)

Thanks.

Note:  If I remove over half of the KEY properties the script runs successfully.

**Comment by duartealexf** [comment]
Hi @delebash 

Thanks for creating this issue. This happens to me as well, with SQL you provided.

TLDR: sorry, no quick fix... use CREATE TABLE first and ALTER TABLE to add the KEYs (depending on which software created this SQL, there might be a flag you can use to do this automatically).

The longer answer...
I'm sorry but there isn't a quick fix for this. Part of the problem lies in the parsing engine I'm using (nearley), which has options to help to fix this but it would require a lot of work, mainly because the lexer part is not well documented.

It would be probably easier to completely replace the parsing engine and update the grammar files, which would take about 2-3 months. I did some study and intend to switching it to the chevrotain parsing engine. I just need the time to actually start it.

Again, I'm really sorry, I wish there was a quick fix, but there just isn't. 😫

**Comment by delebash** [comment]
Thanks for checking on this. Take care!
